stablehlo/reference/ProcessGrid.cpp

```diff
@@ -228,5 +278,19 @@ std::shared_ptr<RendezvousResult const> ProcessGrid::rendezvous(
   return result;
 }
 
+void ProcessGrid::send(ArrayRef<Tensor> inputs, ChannelId channelId,
+                       ProcessId processId) {
+  std::unique_lock<std::mutex> lock(sendRecvChannels_[channelId].mutex);
+  if (!sendRecvConditions_[channelId].wait_for(
+          lock, std::chrono::seconds(3), [&] {
+            return sendRecvReady_.find(channelId) != sendRecvReady_.end();
+          }))
+    llvm::report_fatal_error("send timed out");
+
+  sendRecvChannels_[channelId].result = llvm::to_vector(inputs);
+  sendRecvReady_.erase(channelId);
```

Comment:
Can multiple processes be receiving on the same channel? This setup currently only handles 1:1 communication. If this is how XLA handles these then this is OK

Reply:
There should only be 1:1 communication happening within a channel, otherwise it is undefined.